src/MongoDB/Aggregate/Builder.php

```diff
@@ -76,21 +76,15 @@ public function addStage(array $stage)
     }
 
     /**
-     * Concat an array of stages to the end of the existing pipeline,
-     * will ignore any stages not supported by the builder
+     * Concat an array of stages to the end of the existing pipeline
      * @param array $stages
      */
     public function addStages(Iterable $stages)
     {
-        // We don't want to array_merge, we want to go through and add each stage per our builder. We don't know where these came from.
         foreach ($stages as $stage) {
-            if (!empty($stage)) {
-                $stageName = str_replace('$', '', array_keys($stage)[0]);
-                if (method_exists($this, $stageName)) {
-                    $this->{$stageName}($stage["\${$stageName}"]);
-                }
-            }
+            $this->addStage($stage);

```

Comment:
Since the function was purposely written to validate methods for I assume _safety_ wouldn't it be best to leave it in tact and either substitute a new method or a function argument for force adding stages? ;)